server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -791,16 +848,44 @@ public void deleteStaleSegments(int lastNMetadataFilesToKeep) throws IOException
             return;
         }
 
-        List<String> metadataFilesEligibleToDelete = new ArrayList<>(
-            sortedMetadataFileList.subList(lastNMetadataFilesToKeep, sortedMetadataFileList.size())
+        // Check last fetch status of pinned timestamps. If stale, return.
+        if (RemoteStoreUtils.isPinnedTimestampStateStale()) {
```

Comment:
In `isPinnedTimestampStateStale` , why are we checking for `getPinnedTimestampsLookbackInterval` ?

Reply:
When we pin the timestamp, we can pin till `PinnedTimestampsLookbackInterval` in the past.

That means, if `PinnedTimestampsLookbackInterval=1m`, when we are pinning timestamp at 08:00, we can provide value as 07:59 as well (this is to make sure that we provide some buffer to caller).

As the pinned timestamp value can go back in the past, we are considering the same buffer while checking if pinned timestamp state is stale or not.